internal/adminapi/client_test.go

```diff
@@ -0,0 +1,239 @@
+package adminapi
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+var ca *x509.Certificate
+var caPEM *bytes.Buffer
+var caPrivateKeyPEM *bytes.Buffer
+var certPEM *bytes.Buffer
+var certPrivateKeyPEM *bytes.Buffer
+
+func TestMakeHTTPClient(t *testing.T) {
+
+	buildTLS(t)
+
+	var opts = HTTPClientOpts{
+		true,
+		"",
+		"",
+		caPEM.String(),
+		nil,
+		"",
+		certPEM.String(),
+		"",
+		certPrivateKeyPEM.String(),
+	}
+
+	httpclient, err := MakeHTTPClient(&opts)
+	if err != nil {
+		t.Errorf("Fail to make the HTTP client - %s", err.Error())
+	}
+
+	assert.NotNil(t, httpclient)
+
+	validate(t, httpclient)
+
+}
+
+func TestMakeHTTPClientWithFilePaths(t *testing.T) {
+
+	buildTLS(t)
+
+	caFile, err := ioutil.TempFile("", "ca.crt")
+	if err != nil {
+		t.Errorf("Fail to create CA certificate file - %s", err.Error())
+	}
```

Comment:
As we're already using testify in other areas of this file we can use it here too to check the error to reduce code. Additionally we can use `require` instead of `assert` to halt the tests since no meaningful progress can be made if this file can't be created:

```suggestion
	require.NoError(t, err)
```